integration/typegen-test.ts

```diff
@@ -99,15 +99,15 @@ test.describe("typegen", () => {
           import type { Expect, Equal } from "../expect-type"
           import type { Route } from "./+types/only-optional"
           export function loader({ params }: Route.LoaderArgs) {
-            type Test = Expect<Equal<typeof params, { id?: string }>>
+            type Test = Expect<Equal<typeof params.id, string | undefined>>
```

Comment:
Expanding optionals means that `params` become a union, so can't just check a single object. Can we simplify this at all?

```ts
type x1 = { id?: undefined } | { id: string } | { id: string }
type x2 = { id: string } | { id: string }
```

Reply:
No easy way to do this so bailing on type simplifications